File: api/src/main/java/com/ruby/rubia_server/core/dto/CreateCustomerDTO.java
package com.ruby.rubia_server.core.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CreateCustomerDTO {
    
    @NotBlank(message = "Telefone é obrigatório")
    @Pattern(regexp = "^\\+55\\d{10,11}$", message = "Telefone deve estar no formato +55XXXXXXXXXXX")
    private String phone;
    
    @Size(min = 2, max = 255, message = "Nome deve ter entre 2 e 255 caracteres")
    private String name;
    
    @Size(max = 255, message = "WhatsApp ID não pode exceder 255 caracteres")
    private String whatsappId;
    
    private String profileUrl;
    
    @Builder.Default
    private Boolean isBlocked = false;

    private String sourceSystemName;
    private String sourceSystemId;
    private LocalDateTime importedAt;
    private LocalDate birthDate;
    private LocalDate lastDonationDate;
    private LocalDate nextEligibleDonationDate;
    
    @Size(max = 10, message = "Tipo sanguíneo não pode exceder 10 caracteres")
    private String bloodType;
    
    private Integer height; // Altura em centímetros
    private Double weight; // Peso em quilogramas
    
    // Campos de endereço
    @Size(max = 255, message = "Rua não pode exceder 255 caracteres")
    private String addressStreet;
    
    @Size(max = 20, message = "Número não pode exceder 20 caracteres")
    private String addressNumber;
    
    @Size(max = 255, message = "Complemento não pode exceder 255 caracteres")
    private String addressComplement;
    
    @Size(max = 10, message = "CEP não pode exceder 10 caracteres")
    @Pattern(regexp = "^\\d{5}-?\\d{3}$|^$", message = "CEP deve estar no formato XXXXX-XXX")
    private String addressPostalCode;
    
    @Size(max = 100, message = "Cidade não pode exceder 100 caracteres")
    private String addressCity;
    
    @Size(max = 2, message = "Estado deve ter 2 caracteres")
    @Pattern(regexp = "^[A-Z]{2}$|^$", message = "Estado deve ser uma UF válida (ex: SP, RJ)")
    private String addressState;
    
    // Campos adicionais para campanhas
    @Size(max = 255, message = "Email não pode exceder 255 caracteres")
    private String email;
    
    @Size(max = 14, message = "CPF não pode exceder 14 caracteres")
    private String cpf;
    
    @Size(max = 20, message = "RG não pode exceder 20 caracteres")
    private String rg;
    
    @Size(max = 10, message = "Fator RH não pode exceder 10 caracteres")
    private String rhFactor;
}

File: api-dotnet/Rubia.Server/DTOs/CreateCustomerDto.cs
using System.ComponentModel.DataAnnotations;
using System.Text.RegularExpressions;

namespace Rubia.Server.DTOs;

public class CreateCustomerDto : IValidatableObject
{
    [Required(ErrorMessage = "Telefone é obrigatório")]
    public string Phone { get; set; } = string.Empty;

    [StringLength(255, MinimumLength = 2, ErrorMessage = "Nome deve ter entre 2 e 255 caracteres")]
    public string? Name { get; set; }

    [StringLength(255, ErrorMessage = "WhatsApp ID não pode exceder 255 caracteres")]
    public string? WhatsappId { get; set; }

    public string? ProfileUrl { get; set; }

    public bool IsBlocked { get; set; } = false;

    public string? SourceSystemName { get; set; }

    public string? SourceSystemId { get; set; }

    public DateTime? ImportedAt { get; set; }

    public DateOnly? BirthDate { get; set; }

    public DateOnly? LastDonationDate { get; set; }

    public DateOnly? NextEligibleDonationDate { get; set; }

    [StringLength(10, ErrorMessage = "Tipo sanguíneo não pode exceder 10 caracteres")]
    public string? BloodType { get; set; }

    public int? Height { get; set; } // Altura em centímetros

    public double? Weight { get; set; } // Peso em quilogramas

    // Campos de endereço
    [StringLength(255, ErrorMessage = "Rua não pode exceder 255 caracteres")]
    public string? AddressStreet { get; set; }

    [StringLength(20, ErrorMessage = "Número não pode exceder 20 caracteres")]
    public string? AddressNumber { get; set; }

    [StringLength(255, ErrorMessage = "Complemento não pode exceder 255 caracteres")]
    public string? AddressComplement { get; set; }

    [StringLength(10, ErrorMessage = "CEP não pode exceder 10 caracteres")]
    public string? AddressPostalCode { get; set; }

    [StringLength(100, ErrorMessage = "Cidade não pode exceder 100 caracteres")]
    public string? AddressCity { get; set; }

    [StringLength(2, ErrorMessage = "Estado deve ter 2 caracteres")]
    public string? AddressState { get; set; }

    // Campos adicionais para campanhas
    [StringLength(255, ErrorMessage = "Email não pode exceder 255 caracteres")]
    [EmailAddress(ErrorMessage = "Email deve ter formato válido")]
    public string? Email { get; set; }

    [StringLength(14, ErrorMessage = "CPF não pode exceder 14 caracteres")]
    public string? Cpf { get; set; }

    [StringLength(20, ErrorMessage = "RG não pode exceder 20 caracteres")]
    public string? Rg { get; set; }

    [StringLength(10, ErrorMessage = "Fator RH não pode exceder 10 caracteres")]
    public string? RhFactor { get; set; }

    public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
    {
        var results = new List<ValidationResult>();

        // Validar telefone - formato brasileiro
        if (!string.IsNullOrEmpty(Phone))
        {
            var phoneRegex = new Regex(@"^\+55\d{10,11}$");
            if (!phoneRegex.IsMatch(Phone))
            {
                results.Add(new ValidationResult(
                    "Telefone deve estar no formato +55XXXXXXXXXXX",
                    new[] { nameof(Phone) }));
            }
        }

        // Validar CEP
        if (!string.IsNullOrEmpty(AddressPostalCode))
        {
            var cepRegex = new Regex(@"^\d{5}-?\d{3}$");
            if (!cepRegex.IsMatch(AddressPostalCode))
            {
                results.Add(new ValidationResult(
                    "CEP deve estar no formato XXXXX-XXX",
                    new[] { nameof(AddressPostalCode) }));
            }
        }

        // Validar estado (UF)
        if (!string.IsNullOrEmpty(AddressState))
        {
            var stateRegex = new Regex(@"^[A-Z]{2}$");
            if (!stateRegex.IsMatch(AddressState))
            {
                results.Add(new ValidationResult(
                    "Estado deve ser uma UF válida (ex: SP, RJ)",
                    new[] { nameof(AddressState) }));
            }
        }

        return results;
    }
}